s3-parallel-put

```diff
@@ -127,20 +128,35 @@ class Value(object):
     def should_copy_content(self):
         return self.bucket_name is None
 
+def excluded(pathname, options):
+  for glob in options.include:
+    if fnmatch.fnmatch(pathname, glob):
+      return False
+
+  for glob in options.exclude:
+    if fnmatch.fnmatch(pathname, glob):
+      return True
+
+  return False
 
 def walk_filesystem(source, options):
     if os.path.isdir(source):
         for dirpath, dirnames, filenames in os.walk(source):
+            if excluded(dirpath, options):
+                continue
             for filename in filenames:
                 abs_path = os.path.join(dirpath, filename)
                 if not os.path.isfile(abs_path):
                     continue
+                if excluded(filename, options):
+                    continue
                 rel_path = os.path.relpath(abs_path, source)
                 key_name = '/'.join([options.prefix] + rel_path.split(os.sep))
                 yield (key_name, dict(path=abs_path))
     elif os.path.isfile(source):
-        key_name = os.path.normpath(os.path.join(options.prefix, source))
-        yield (key_name, dict(path=source))
+        if not excluded(source, options):
```

Comment:
Can you use the `if excluded(source, options): continue` pattern used in line 145 above here? I think it would make the code cleaner.